### ISSUE TITLE
consider adding support for aria-pressed to calcite-button

### ISSUE THREAD

**Issue by clhenrick** [Esri/calcite-components#13655]
**monday.com sync:** #10938340729

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Button

### Summary

In order to implement a calcite-button as a [toggle button,](https://dequeuniversity.com/library/aria/button-toggle) the [aria-pressed](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-pressed) attribute needs to be added to support assistive tech.

### Actual Behavior

Adding the `aria-pressed="true|false"` attribute to a calcite-button does not pass through the component's Shadow DOM and so is not applied to the inner button element. This means assistive tech likely won't pick it up when encountering it.

### Expected Behavior

The `aria-pressed` attribute should be passed down to the inner button element when added as an attribute to the host.

### Reproduction Sample

https://codepen.io/clhenrick/pen/ogLxOoe?editors=1111

### Reproduction Steps

1. create an instance of calcite-button
2. add the attribute `aria-pressed="false"`
3. observe that the button in the component's Shadow DOM does not get the attribute

### Reproduction Version

5.0

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Instant Apps
